php_phongo.c

```diff
@@ -466,10 +519,14 @@ static int process_read_preference(zval *option, zval **zreadPreference TSRMLS_D
 	return true;
 }
 
-static int process_write_concern(zval *option, zval **zwriteConcern TSRMLS_DC)
+static int process_write_concern(zval *option, bson_t *mongoc_opts, zval **zwriteConcern TSRMLS_DC)
 {
 	if (Z_TYPE_P(option) == IS_OBJECT && instanceof_function(Z_OBJCE_P(option), php_phongo_writeconcern_ce TSRMLS_CC)) {
+		const mongoc_write_concern_t *write_concern = phongo_write_concern_from_zval(option TSRMLS_CC);
+
 		*zwriteConcern = option;
+
+		mongoc_write_concern_append((mongoc_write_concern_t*) write_concern, mongoc_opts);
```

Comment:
`mongoc_write_concern_append()` returns a success boolean. If false, you should throw an exception saying we couldn't append it properly.

Reply:
done